sklearn_numba_dpex/common/random.py

```diff
@@ -0,0 +1,307 @@
+    # block should be used instead, and the tests should be adapted. Else, the block
+    # can be removed.
+
+    # jump_const_1 = uint64(0xDF900294D8F554A5)
+    # jump_const_2 = uint64(0x170865DF4B3201FC)
+    # jump_const_3 = uint64(1)
+    # jump_init = uint64(0)
+    # long_2 = int64(2)
+    # long_64 = int64(64)
+
+    # @dpex.func
+    # def _xoroshiro128pp_jump(states, index):
+    #     """Advance the RNG in ``states[index]`` by 2**64 steps."""
+    #     s0 = jump_init
+    #     s1 = jump_init
+
+    #     for i in range(long_2):
+    #         if i == zero_idx:
+    #             jump_const = jump_const_1
+    #         else:
+    #             jump_const = jump_const_2
+    #         for b in range(long_64):
+    #             if jump_const & jump_const_3 << uint32(b):
+    #                 s0 ^= states[index, zero_idx]
+    #                 s1 ^= states[index, one_idx]
+    #             _xoroshiro128p_next(states, index)
+
+    #     states[index, zero_idx] = s0
+    #     states[index, one_idx] = s1
+
+    # Initialization is purely sequential so it will be faster on CPU, if a cpu device
+    # is available make sure to use it.
+    if device is None:
+        device = dpctl.SyclDevice()
+    from_cpu_to_device = False
+    if not device.has_aspect_cpu:
+        try:
+            cpu_device = dpctl.SyclDevice("cpu")
+            from_cpu_to_device = True
+        except dpctl.SyclDeviceCreationError:
+            warnings.warn(
+                "No CPU found, fallbacking random initiatlization to default " "device."
+            )
+
+    states = dpt.empty(
+        sh=(n_states, 2),
+        dtype=np.uint64,
+        device=cpu_device if from_cpu_to_device else device,
+    )
+    init_xoroshiro128pp_states[1, 1](states)
+
+    if from_cpu_to_device:
+        return states.to_device(device)
+
+    else:
+        return states
+
+
+next_rot_1 = uint32(17)
+next_rot_2 = uint32(49)
+next_rot_3 = uint32(28)
+shift_1 = uint32(21)
+
+
+@dpex.func
+def _xoroshiro128pp_next(states, index):
+    """Return the next random uint64 and advance the RNG in states[index]."""
+    s0 = states[index, zero_idx]
+    s1 = states[index, one_idx]
+    result = _rotl(s0 + s1, next_rot_1) + s0
+
+    s1 ^= s0
+    states[index, zero_idx] = _rotl(s0, next_rot_2) ^ s1 ^ (s1 << shift_1)
+    states[index, one_idx] = _rotl(s1, next_rot_3)
+
+    return result
+
+
+# next_rot_p_1 = uint32(24)
+# next_rot_p_2 = uint32(16)
+# next_rot_p_3 = uint32(37)
+
+
+# @dpex.func
+# def _xoroshiro128p_next(states, index):
+#     """Return the next random uint64 and advance the RNG in states[index]."""
+#     s0 = states[index, zero_idx]
+#     s1 = states[index, one_idx]
+#     result = s0 + s1
+
+#     s1 ^= s0
+#     states[index, zero_idx] = _rotl(s0, next_rot_p_1) ^ s1 ^ (s1 << next_rot_p_2)
+#     states[index, one_idx] = _rotl(s1, next_rot_p_3)
+
+#     return result
+
+
+uint32_64 = uint32(64)
```

Comment:
To me, this naming is clearer.
```suggestion
64_as_uint32 = uint32(64)
```